float64 x
float64 y
float64 z
float64 t1
float64 t2
float64 t3
---
float64 q1
float64 q2
float64 q3